src/fixed_ride_type.h

```diff
@@ -21,7 +21,12 @@ class FixedRideType : public RideType {
 	explicit FixedRideType(RideTypeKind kind);
 	~FixedRideType();
 
-	virtual int GetRideCapacity() const = 0;
+	/* Information about how many guests can use the ride at the same time. */
+	struct RideCapacity {
+		int number_of_batches; ///< How many batches of guests fit into the ride.
+		int guests_per_batch;  ///< How many guests may be in each batch.
```

Comment:
Shouldn't there be 2 values for "number of guests in the ride" in the FGTR block?

Please keep the block definitions and code in sync.

Reply:
A fixed ride can only ever contain 1 batch of people, so there's no need to put this in the RCD imho. The only two exceptions I can think of in the original RCT are the ferris wheel, which is such a *special* ride in every way that it will likely need its own class when (if?) it gets implemented, and Tracked Rides (maze, gokarts, etc), which are technically a coaster derivate and also not covered by FixedRideType.

Reply:
I usually don't stop where the original RCT ran out of time or money.

The two examples I had in mind for this were all coaster rides, and a movie theater where several movies can be shown at the same time. Your haunted house may also be an example. I can imagine several small groups of people visit the ride at the same time, one group after the other.

Not sure why you consider the Ferris Wheel special? Just make an animation to turn the wheel to the next cabin (which could be eg 11/6 further down the wheel to fake the idea of turning the wheel between changing passengers). As long as all cabins get a turn to exchange passengers, it's fine. (Where 11/6 is just  an example, didn't verify all cabins actually get a turn then.)

Reply:
Actually, for `N` cabins, the simplest is to turn `N-1` cabins or `N+1` cabins, which trivially gives all cabins a turn.

Reply:
Okay, I've added this to the RCD, and also split up the spooky house's group. However I had to add the requirement that rides with more than 1 batch mustn't have a real working animation, as then the ride's idle/working phases (and the point in time in the working animation) are no longer well-defined.

For the ferris wheel, this implementation would be possible though it would have the weird side-effect that each guest would first stand on the ground waiting for a cabin, then ride one round, then stand around on the ground waiting for the remaining guests to exit. I would expect that he'd spend the first waiting time in the queue (not enter the ride until a free cabin is on the ground) and exit the ride immediately when exiting the cabin. But I guess that this could be considered an eye-candy detail so ok…

Reply:
Current decisions are good enough for now, I think.

I don't see "idle" and "empty" as the same things, although it holds for a large number of rides. Also "working" and loading/unloading are perhaps less mutual exclusive than currently assumed, eg toilets with finite space inside.
But it's all for the future.